### PR TITLE
Fix Scala loop

### PR DESCRIPTION
**Pull request by He-Pin**
I have:

* [ x] Read the project [README](../README.md), including the [benchmark descriptions](../README.md#available-benchmarks)

## Description of changes

refs: https://github.com/bddicken/languages/issues/304

Modification:
1. remove the scala-js jitless test, which will not pass
2. `reset the j`, it was wrong.

### CODE REVIEW

**Comment by He-Pin** [comment]
@PEZ I think this is now fixed.

File: loops/scala/base.scala [hunk 1]
```diff
@@ -3,12 +3,13 @@
...
-  def run(number: String): Unit =
-    val u = number.toInt // Get an input number from the command line
+  def run(args: Array[String]): Unit =
+    val u = args(0).toInt // Get an input number from the command line
     val r = Random().nextInt(10000) // Get a random number 0 <= r < 10k
     val a = new Array[Int](10000) // Array of 10k elements initialized to 0
-    var i, j = 0
+    var i = 0
     while i < 10000 do // 10k outer loop iterations
+      var j = 0
```
Comment (He-Pin) [inline loops/scala/base.scala:12]: `j` should be reset to `0`

**Comment by He-Pin** [comment]
image
```scala
hepin@Mac loops % ../compile.sh

Compiling scala

Compiling scala

Compiling scala
Closure: 0 error(s), 0 warning(s)

Compiling scala
  [37ms]  bundle  1 modules
  [94ms] compile  scala/bun
hepin@Mac loops % ../run.sh    

Checking Scala
Check passed
Benchmarking Scala
Benchmark 1: ./scala/code 40
  Time (mean ± σ):     616.7 ms ±   0.4 ms    [User: 607.9 ms, System: 26.7 ms]
  Range (min … max):   616.2 ms … 617.0 ms    3 runs
 
  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet system without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.
 

Checking Scala-Native
Check passed
Benchmarking Scala-Native
Benchmark 1: ./scala/code-native 40
  Time (mean ± σ):     510.8 ms ±   0.2 ms    [User: 507.9 ms, System: 1.7 ms]
  Range (min … max):   510.5 ms … 511.0 ms    3 runs
 

Checking Bun Scala-JS(Compiled)
Check passed
Benchmarking Bun Scala-JS(Compiled)
Benchmark 1: ./scala/bun 40
  Time (mean ± σ):     961.0 ms ±  15.0 ms    [User: 970.0 ms, System: 8.8 ms]
  Range (min … max):   949.4 ms … 977.9 ms    3 runs
 

Checking Bun Scala-JS
Check passed
Benchmarking Bun Scala-JS
Benchmark 1: bun ./scala/code.js 40
  Time (mean ± σ):     962.6 ms ±   7.5 ms    [User: 973.2 ms, System: 8.9 ms]
  Range (min … max):   955.3 ms … 970.3 ms    3 runs

```

**Comment by He-Pin** [comment]
The `for ()` is extremely slow for scala-native:
```scala
object Loop {
  def run(args: Array[String]): Unit = {
    val u = args(0).toInt // Get an input number from the command line
    val r = new Random().nextInt(10000) // Get a random number 0 <= r < 10k
    val a = new Array[Int](10000) // Array of 10k elements initialized to 0
    for (i <- 0 until 10000) { // 10k outer loop iterations
      for (j <- 0 until 100000) { // 100k inner loop iterations, per outer loop iteration
        a(i) = a(i) + j % u // Simple sum
      }
      a(i) += r // Add a random value to each element in array
    }
    println(a(r)) // Print out a single element from the array
  }
}
```

and then 5 seconds, slower than scala-js:
```scala
hepin@Mac loops % ../run.sh

Checking Scala
Check passed
Benchmarking Scala
Benchmark 1: ./scala/code 40
  Time (mean ± σ):     667.6 ms ±   5.1 ms    [User: 661.2 ms, System: 26.5 ms]
  Range (min … max):   661.7 ms … 671.2 ms    3 runs
 

Checking Scala-Native
Check passed
Benchmarking Scala-Native
Benchmark 1: ./scala/code-native 40
  Time (mean ± σ):      5.313 s ±  0.039 s    [User: 5.298 s, System: 0.014 s]
  Range (min … max):    5.282 s …  5.356 s    3 runs
 

Checking Bun Scala-JS(Compiled)
Check passed
Benchmarking Bun Scala-JS(Compiled)
Benchmark 1: ./scala/bun 40
  Time (mean ± σ):     962.2 ms ±   7.3 ms    [User: 973.0 ms, System: 8.1 ms]
  Range (min … max):   957.6 ms … 970.7 ms    3 runs
 

Checking Bun Scala-JS
Check passed
Benchmarking Bun Scala-JS
Benchmark 1: bun ./scala/code.js 40
  Time (mean ± σ):     970.0 ms ±   9.8 ms    [User: 980.2 ms, System: 7.8 ms]
  Range (min … max):   959.4 ms … 978.8 ms    3 runs

```

**Comment by PEZ** [comment]
That `for()` impact is a huge mystery!

**Comment by PEZ** [comment]
Thanks for fixing this! ❤️

**Comment by He-Pin** [comment]
Thanks for merging, would you like to update your visual result page:) , I would like to see Scala been included there. @PEZ

**Pull request merged**